api/data_explorer/util/elasticsearch_util.py

```diff
@@ -215,13 +240,20 @@ def get_elasticsearch_facet(es, elasticsearch_field_name, field_type):
             field=elasticsearch_field_name,
             interval=_get_bucket_interval(field_range))
 
-    # Handle sample facets in a special way since they are nested objects.
-    if elasticsearch_field_name.startswith('samples.'):
-        es_facet = NestedFacet('samples', es_facet)
+    nested_path = is_nested_facet(elasticsearch_field_name, nested_facet_paths)
+    if nested_path:
+        es_facet = NestedFacet(nested_path, es_facet)
 
     return es_facet
 
 
+def is_nested_facet(elasticsearch_field_name, nested_facet_paths):
```

Comment:
Internal methods begin with _
By convention, methods are defined before they're used. Please move to before get_elasticsearch_facet().
is_nested_facet() sounds like it returns a boolean. How about:
```
def _maybe_get_nested_path
"""
If given field is a nested field, returns nested path. Otherwise returns None.
"""
```